src/manager.cpp

```diff
@@ -203,12 +203,109 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    nlohmann::json l_sysCfgJsonObj{};
+    int l_bytesWritten = -1;
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    if (m_worker.get() != nullptr)
+    {
+        l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    try
+    {
+        // For the given VPD path, get it's corresponding inventory object path
+        // from system config JSON
+        const std::string& l_inventoryObjPath =
+            jsonUtility::getInventoryObjPathFromJson(l_sysCfgJsonObj,
+                                                     i_vpdPath);
+
+        // For the given VPD path, get it's corresponding hardware path from
+        // system config JSON.
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_vpdPath);
+
+        // If given VPD path is not found in JSON or if system config JSON
+        // object is empty
+        if ((l_inventoryObjPath.empty() && l_fruPath.empty()) ||
+            l_sysCfgJsonObj.empty())
+        {
+            // Perform write on hardware
+            return (writeOnHardware(i_vpdPath, l_sysCfgJsonObj,
+                                    i_paramsToWriteData));
+        }
+
+        types::DbusPropertyName l_propertyName;
+        types::DbusVariantType l_propertyValue;
+        types::DbusInterface l_interfaceName;
+
+        // Extract data from input
+        if (const types::IpzData* l_ipzData =
+                std::get_if<types::IpzData>(&i_paramsToWriteData))
+        {
+            types::Record l_recordName = std::get<0>(*l_ipzData);
+            l_interfaceName = constants::ipzVpdInf + l_recordName;
+
+            l_propertyName = std::get<1>(*l_ipzData);
+
+            types::BinaryVector l_keywordData = std::get<2>(*l_ipzData);
+            l_propertyValue.emplace<types::BinaryVector>(l_keywordData);
+        }
+        else
+        {
+            logging::logMessage(
+                "Input parameter type provided isn't compatible with the given VPD type.");
+            throw types::DbusInvalidArgument();
+        }
+
+        // Update the keyword's value on D-bus inventory object path.
+        int l_bytesWrittenOnDbus = writeOnDbus(l_inventoryObjPath,
+                                               l_interfaceName, l_propertyName,
+                                               l_propertyValue);
+
+        // Update the keyword's value on hardware path.
+        int l_bytesWrittenOnHardware =
+            writeOnHardware(l_fruPath, l_sysCfgJsonObj, i_paramsToWriteData);
```

Comment:
No need to call this API twice as this is a part of common flow irrespective of Json object present or absent. This case be optimized by keeping the call to writeHardware common and just copying the fru path to vpd path in case coming through Json flow.

Reply:
1. I want the input variable i_vpdPath to be constant.
2. Also, as we discussed, first to update on dbus then only on hardware. so that in case a dbus failure happens, we can stop updating the hardware.. (I have a todo section to handle the exceptional cases.)

so I would suggest to have the code flow as it is.